### PR TITLE
Fixed and Refactored Gravity Well

### PR DESCRIPTION
**Pull request by DJSteins**

### CODE REVIEW

**Comment by DJSteins** [comment]
[Gravity Well doesn't pull car and messes up controls](https://trello.com/c/MkhJLtrI/442-gravity-well-doesnt-pull-car-and-messes-up-controls)

File: Assets/_PartyParkade/Script/PowerUp/GravityWell.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+//     Copyright (c) PERIGON GAMES. 2019 All Rights Reserved
+// </copyright>
+//-----------------------------------------------------------------------
+using System.Collections.Generic;
+using _PartyParkade.Script.Data;
+using _PartyParkade.Script.Game;
+using _PartyParkade.Script.Game.Car;
+using UnityEngine;
+
+public class GravityWell
```
Comment (m4rtint) [inline Assets/_PartyParkade/Script/PowerUp/GravityWell.cs]: Formatting stuff
- comments, protectors

File: Assets/_PartyParkade/Script/PowerUp/GravityWellBehaviour.cs [hunk 2]
```diff
@@ -100,11 +71,11 @@ private void OnTriggerEnter(Collider other)
...
     {
-        Rigidbody rg = other.GetComponent<Rigidbody>();
-        if (m_CarsToPull.Contains(rg))
+        PlayerController playerController = other.GetComponentInParent<PlayerController>();
+        CarController carController = playerController.Car;
+        if (m_CarsToPull.Contains(carController))
         {
-            m_CarsToPull.Remove(rg);
-            m_lengthOfCarsList--;
+            m_CarsToPull.Remove(carController);
```
Comment (Gursaanj) [inline Assets/_PartyParkade/Script/PowerUp/GravityWellBehaviour.cs]: Not sure you have to do a contains check, cause Remove already does a contains check internally and wont do anything if m_CarsToPull does not contain carController

File: Assets/_PartyParkade/Script/PowerUp/GravityWellBehaviour.cs [hunk 3]
```diff
@@ -11,85 +11,56 @@
...
-            m_currentPlayerNumber = value;
-        }
-    }
+    private float m_strengthOfPull = 1000.0f;
 
     private void Awake()
     {
-        m_CarsToPull = new List<Rigidbody>();
-        m_centerpositionTransform = this.transform;
+        m_CarsToPull = new List<CarController>();
```
Comment (Gursaanj) [inline Assets/_PartyParkade/Script/PowerUp/GravityWellBehaviour.cs]: it should be m_carsToPull

**Pull request merged**